### DISCUSSION TITLE
Fix for viewport height change on Chrome on Android

### DISCUSSION THREAD

**Discussion by Zwyx** [vercel/next.js#63724] [Show and tell] [answered]
@DustinBrett and @icyJoseph, I've noticed your discussion about the viewport height change in issue https://github.com/vercel/next.js/issues/47839

I've also encountered this issue, with the React-TypeScript Vite template.

I just wanted to let you know that I found a fix: adding `interactive-widget=resizes-content` to the `viewport` meta tag.

- This page has the problem: https://zwyx.github.io/chrome-android-clientheight/simple
- This page is fixed: https://zwyx.github.io/chrome-android-clientheight/simple-fixed

Read more about `interactive-widget` here: https://developer.chrome.com/blog/viewport-resize-behavior

I still think this is a bug in Chrome, so I reported it. Read more at https://github.com/Zwyx/chrome-android-clientheight

**Comment by DustinBrett** [comment]
Thanks for this info. I will look into `interactive-widget`. My solution using `beforeInteractive` and tracking initial `window.innerHeight` is afaik working still, but I would be happy for a cleaner solution using meta tags.

**Reply by Zwyx** [reply]
Just letting you know that this was indeed a bug in Chrome, which is fixed in Chrome 139. See https://issues.chromium.org/issues/331326389

**Comment by m-a-r-c-e-l-i-n-o** [comment]
@DustinBrett How are you getting the correct value by using `beforeInteractive`? For me, the height is wrong from the get-go. Doing `<Script strategy="beforeInteractive">{'console.log('window.innerHeight', window.innerHeight)'}</Script>` inside the `_document.tsx` file yields the incorrect value still.

**Reply by DustinBrett** [reply]
For me that initial value was the correct one most of the time, but not 100%.

I was doing this after `<Head />` inside the start of `<body>`

```jsx
<Script id="initHeight" strategy="beforeInteractive">
  window.initialHeight = window.innerHeight;
</Script>
```

But I have since moved to `interactive-widget=resizes-content` which fixed things. Although it required me setting my main viewport element to `overflow: clip` otherwise the virtual keyboard could sometimes mess up elements which were set to `bottom: 0`.

**Reply by m-a-r-c-e-l-i-n-o** [reply]
Got it :) thanks for clarifying! Out of curiosity, are you dealing with Android and iOS separately? I ask this because `interactive-widget=resizes-content` will change the layout behavior on Android, but iOS will continue to behave like `interactive-widget=resizes-visual`.

**Reply by DustinBrett** [reply]
I was only dealing with the issue on my own Android device as I noticed the problem when I opened my site in Incognito after having the virtual keyboard open right before the page opened.

I couldn't recreate the issue on my site with iOS yet, but I only have an older iPhone. I was hoping my issue was just Chromium related.

**Reply by m-a-r-c-e-l-i-n-o** [reply]
Oh no, the issue is much more robust and more readily reproducible (happens without incognito), but it's only Android (did a lot of testing on BrowserStack). My mention of iOS was more about the behavior of the keyboard that changes when you apply `interactive-widget=resizes-content`. On iOS the content will not resize, meanwhile on Android it will resize, this forces us into a dual strategy on how to command the layout when the keyboard opens.

**Comment by robertcx** [comment]
Hello. First of all, thank you for finding a solution to this. I've recently picked up NextJS and it has bugged me since the very beginning that something so trivial could be so hard to get fixed. I've luckily stumbled on your post and it has helped me get it working as it should. Just wanted to share how I did it in case other beginners like me struggle with implementing it in Next.

I've added the following code to my `layout.tsx` file.

```
import type { Viewport } from 'next';

export const viewport: Viewport = {
  interactiveWidget: 'resizes-content',
};
```

Upon checking in DevTools, it is exactly as it should be within the meta-tag. And nevertheless, it works perfectly. Thanks again for this.

**Comment by Zwyx** [comment, marked as answer]
This was indeed a bug in Chrome, which is fixed in Chrome 139. See https://issues.chromium.org/issues/331326389
